src/proteus/dswx_hls.py

```diff
@@ -3373,22 +3461,23 @@ def generate_dswx_layers(input_list,
 
     diagnostic_layer_decimal = _compute_diagnostic_tests(
         blue, green, red, nir, swir1, swir2, hls_thresholds)
- 
     diagnostic_layer_decimal[invalid_ind] = DIAGNOSTIC_LAYER_NO_DATA_DECIMAL
 
     interpreted_dswx_band = generate_interpreted_layer(
         diagnostic_layer_decimal)
-    
     diagnostic_layer = _get_binary_representation(diagnostic_layer_decimal)
     del diagnostic_layer_decimal
 
     if output_diagnostic_layer:
+        diagnostic_layer_ctable = _get_diagnostic_layer_ctable(
+            diagnostic_layer)
```

Comment:
For documentation purposes: Using `gdalinfo`, discovered that the new DIAG file contains 65535 RGB colortable values, while the original DIAG file did not contain a colortable at all. Here are a couple of other significant differences:

Original DIAG File:
```
Band 1 Block=512x512 Type=UInt16, ColorInterp=Gray
  NoData Value=100000
```

DIAG File generated with this code:
```
Band 1 Block=512x512 Type=UInt16, ColorInterp=Palette
NoData Value=65535
```

Again, I do think it would be *really* awesome to have a colortable for the DIAG layer. Hopefully there's a way to keep the file size, etc. smaller...?

Reply:
Thank you again for catching this, @nemo794 . I just removed the color table for the DIAG layer =/